src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/annotations/PostConstructReturnTypeQuickFix.java

```diff
@@ -33,19 +42,57 @@
  * @author Yijia Jing
  *
  */
-public class PostConstructReturnTypeQuickFix {
+public class PostConstructReturnTypeQuickFix implements IJavaCodeActionParticipant {
+
+    private static final Logger LOGGER = Logger.getLogger(PostConstructReturnTypeQuickFix.class.getName());
+    private final static String TITLE_MESSAGE = Messages.getMessage("ChangeReturnTypeToVoid");
+
+    @Override
+    public String getParticipantId() {
+        return PostConstructReturnTypeQuickFix.class.getName();
+    }
+
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
         List<CodeAction> codeActions = new ArrayList<>();
-        PsiElement node = context.getCoveredNode();
-        PsiMethod parentType = getBinding(node);
-        String name = Messages.getMessage("ChangeReturnTypeToVoid");
-        ChangeCorrectionProposal proposal = new ModifyReturnTypeProposal(name, context.getSource().getCompilationUnit(),
-                context.getASTRoot(), parentType, 0, PsiPrimitiveType.VOID);
-        CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
-        codeActions.add(codeAction);
+        final PsiElement node = context.getCoveredNode();
+        final PsiMethod parentType = getBinding(node);
+
+        if (parentType != null) {
+            codeActions.add(createCodeAction(context, diagnostic));
+        }
         return codeActions;
     }
 
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
+        final PsiElement node = context.getCoveredNode();
+        final PsiMethod parentType = getBinding(node);
+
+        assert parentType != null;
+        ChangeCorrectionProposal proposal = new ModifyReturnTypeProposal(TITLE_MESSAGE, context.getCompilationUnit(),
```

Comment:
The parameter expected here is the source `PsiFile`. This should be retrieved from the Context object in the same way as before. Please refer to the original code.